### DISCUSSION TITLE
Return record from custom function

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [pocketbase/pocketbase#3874] [Q&A]
I need to write a custom function to update a record's json value after checking certain condition in backend. I want to return same record as response. 
Usual case I used to return json value as return c.JSON(200, response) how to return record after update as result?

**Comment by ganigeorgiev** [comment]
It is difficult for me to understand what you are asking and trying to do without a code sample.

What is `response` in the above?
Does it contain the updated record?
How do you fetch and update the record in the first place, aka. is it an existing pointer or a newly created one? If the latter you'll have to replace the response with the new record value.

If the above doesn't help, please elaborate and provide a more detailed code sample.

**Reply by iamdineshkumar** [reply]
Params for the custom function will be id & worktype of a job
Logic is a collection with job with lot of different worktype and its status will be in json format in each job. Based on the entry made towards job worktype status will be updated as progress(I am achieve this through after create hook of job entry collection). Supervisor will check and close specific worktype alone as completed. Since json has multiple worktype and it can be modified anytime through hook as mentioned, I need to achive it through custom function which will check everything and will update the job record json value and will save it.

```



e.Router.AddRoute(echo.Route{
	Method: http.MethodPost,
	Path:   "/api/worksectioncose",
	Handler: func(c echo.Context) error {

		jsonBody := make(map[string]interface{})
		err := json.NewDecoder(c.Request().Body).Decode(&jsonBody)
		if err != nil {
			return apis.NewBadRequestError("Invalid Request", err)
		}
		id := cast.ToString(jsonBody["job"])
		workType := cast.ToString(jsonBody["work"])
		job, err := app.Dao().FindRecordById("jobs", id)
		if err != nil {
			return apis.NewBadRequestError("Failed to fetch job", err)
		}
		workWise, err := appdto.ConvertJOBSectionWise(pc.GetString("workWiseStatus"))
		if err != nil {
			return apis.NewBadRequestError("Failed to parse items", err)
		}

		for i, work := range workWise {
			if work .Section == workType {
				work .Status = "Closed"
				work .StartDate = time.Now().UTC().Format("2006-01-02T15:04:05-0700")
				workWise[i] = work 
				job.Set("workWiseStatus", workWise)
				break
			}
		}
		if err := app.Dao().SaveRecord(job); err != nil {
			return apis.NewBadRequestError("Failed to mark status", err)
		}

	return c.JSON(200, job) //---> i need to send back record over here and in dart lib i need to handle this as usual response and get RecordModel
}
```

**Reply by ganigeorgiev** [reply]
The above code sample seems incomplete and unfortunately doesn't help understanding what is the desired behavior.

In the above code you don't save anywhere the `job` record, only `pc` which is nowhere defined in the sample.

I doubt that this is what you are asking, but if you want to return a record as response it should be enough to call `c.JSON(200, yourRecord)`, but again you are not persisting anywhere in the above the fetched `job`.

Please try to provide as much details as possible with your questions to avoid the needless back-and-forth.

**Reply by iamdineshkumar** [reply]
@ganigeorgiev I have updated complete code. kindly check and let me know.

**Reply by ganigeorgiev** [reply]
This is really frustrating and I don't know what Dart has to do with your custom function, but I'll try one last time and will stop engaging further.

Again, from the Go side returning `c.JSON(200, job)` should work.
For reading the request body, see also https://pocketbase.io/docs/go-routing/#reading-request-body.

If you are asking how to send request from the Dart SDK and get a `RecordModel` value, you can try:

```dart
final record = pb.send("/api/worksectioncose",
  method: "POST",
  body: {...},
).then((value) => RecordModel.fromJson(value as Map<String, dynamic>? ?? {}));
```

**Reply by iamdineshkumar** [reply]
Thanks I will handle based on your recommendation. Actually my question went wrong I think, I have partial memory as somewhere you have mentioned that based on response format from custom function Dart SDK will under stand and return record as in pb.collection("collectionname").insert(body) / getList() functions. That's what I was trying to ask you But I may be wrong. Sorry for bothering you..
